mvlearn/decomposition/baseica.py

```diff
@@ -0,0 +1,233 @@
+# MIT License
+
+# Copyright (c) [2020] [Pierre Ablin and Hugo Richard]
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import numpy as np
+from scipy import linalg
+from sklearn.decomposition import PCA
+
+from .base import BaseDecomposer
+from ..preprocessing.repeat import ViewTransformer
+
+
+class BaseICA(BaseDecomposer):
```

Comment:
Why do we need a BaseICA class? I think that in the short term, we only have GroupICA and multiviewICA implemented in mvlearn, and both work quite differently: groupICA estimates one source matrix from the whole dataset, while multiviewICA outputs multiple sources. 

So I don't think it makes things easier to have a BaseICA class.

Reply:
Also, if we really need a BaseICA class, then we should make GroupICA inherit from it.

Reply:
Ok I agree let's remove PermICA from mvlearn and merge BaseICA and multiViewICA.